### ISSUE TITLE
Assert.Multiple does not provide correct linenumber of failed assertion

### ISSUE THREAD

**Issue by DMDTT** [nunit/nunit#4242]
If only on assertion within an ```Àssert.Multiple```-block fails

```
  using NUnit.Framework;
  
  namespace NunitStack
  {
      public class TestClass
      {
          [Test]
          public void Do1()
          {
              Assert.Multiple(() =>
              {
                  Assert.AreEqual(1, 2);
                  Assert.AreEqual(2, 2);
              });
          }

          [Test]
          public void Do2()
          {
              Assert.Multiple(() =>
              {
                  Assert.AreEqual(1, 2);
                  Assert.AreEqual(2, 3);
              });
          }
      }
  }
```

Test Do1() results in 
```  
  Expected: 1
  But was:  2

   at NunitStack.TestClass.Do1() in C:\Projects\Playground\NunitStack\NunitStack\Class1.cs:line 10
```

TestDo2() results in 
```
  Multiple failures or warnings in test:

  1)   Expected: 1
  But was:  2
   at NunitStack.TestClass.<>c.<Do2>b__1_0() in C:\Projects\Playground\NunitStack\NunitStack\Class1.cs:line 22
   at NUnit.Framework.Assert.Multiple(TestDelegate testDelegate)
   at NunitStack.TestClass.Do2() in C:\Projects\Playground\NunitStack\NunitStack\Class1.cs:line 20

  2)   Expected: 2
  But was:  3
   at NunitStack.TestClass.<>c.<Do2>b__1_0() in C:\Projects\Playground\NunitStack\NunitStack\Class1.cs:line 23
   at NUnit.Framework.Assert.Multiple(TestDelegate testDelegate)
   at NunitStack.TestClass.Do2() in C:\Projects\Playground\NunitStack\NunitStack\Class1.cs:line 20


   at NunitStack.TestClass.Do2() in C:\Projects\Playground\NunitStack\NunitStack\Class1.cs:line 20
```

This is really annoying as if many assertions are within a block and only on assertion fails, you don't know where.
I propose to also printout the stacktrace pointing to a single failing assert.

**Comment by stevenaw** [comment]
Thanks for this report @DMDTT 
Can you confirm how you are running your tests (nunit console, the visual studio adapter, resharper, etc) and any relevant version info?

**Comment by DMDTT** [comment]
of course, sry...

I run the tests via dotnet test cli/ visualstudio testrunner/ resharper unittest sessions, all with the same result.

here my csproj vor my project

```
<PropertyGroup>
    <TargetFramework>net6.0</TargetFramework>
    <ImplicitUsings>enable</ImplicitUsings>
    <Nullable>enable</Nullable>
  </PropertyGroup>

  <ItemGroup>
    <PackageReference Include="Microsoft.NET.Test.Sdk" Version="17.3.2" />
    <PackageReference Include="NUnit" Version="3.13.3" />
    <PackageReference Include="NUnit3TestAdapter" Version="4.2.1" />
  </ItemGroup>
```

**Comment by CharliePoole** [comment]
This looks fixed in 4.1

**Comment by stevenaw** [comment]
I've also been unable to reproduce in 4.x
